source/k64f_eth_nanostack_port.c

```diff
@@ -501,236 +361,275 @@ static int8_t arm_eth_phy_k64f_address_write(phy_address_type_e address_type, ui
+ *      in Hardware registers.
+ *
+ *  \param[in] mac_addr Pointer to MAC address.
+ *
+ */
+static void k64f_eth_initialize(uint8_t *mac_addr)
 {
-    bool link_status = false;
-    enet_dev_if_t *ethernet_iface_ptr =
-            &ethernet_iface[BOARD_DEBUG_ENET_INSTANCE];
-
-    /* First check, if the cable is connected or not.*/
-    phy_get_link_status(ethernet_iface_ptr, &link_status);
-
-    if (!link_status && last_known_status==LINK_STATUS_UP) {
-        driver_readiness_status_callback(0, eth_interface_id);
-        last_known_status = LINK_STATUS_DOWN;
-        ethernet_iface_ptr->isInitialized = false;
-        tr_debug("Ethernet phy link status = down");
-        tr_info("Ethernet Cable is not connected.");
+    uint32_t sysClock;
+    phy_speed_t phy_speed;
+    phy_duplex_t phy_duplex;
+    uint32_t phyAddr = 0;
+
+    enet_config_t config;
+
+
+    /* Allocate TX buffer descriptors */
+    void *memptr = MEM_ALLOC(
+            sizeof(enet_tx_bd_struct_t) * ENET_TX_RING_LEN + ENET_BUFF_ALIGNMENT);
+    if (!memptr) {
+        tr_error("Could not allocate memory for TX Buf Descriptors.");
+        return;
+    }
+    base_DS.tx_buf_desc_start_addr =
+            (volatile enet_tx_bd_struct_t *) ENET_ALIGN(memptr, ENET_BUFF_ALIGNMENT);
+    memset((enet_tx_bd_struct_t *) base_DS.tx_buf_desc_start_addr, 0,
+           sizeof(enet_tx_bd_struct_t) * ENET_TX_RING_LEN);
+
+    /* Allocate RX buffer descriptors */
+    memptr = MEM_ALLOC( sizeof(enet_rx_bd_struct_t) * ENET_RX_RING_LEN + ENET_BUFF_ALIGNMENT);
+    if (!memptr) {
+        tr_error("Could not allocate memory for RX Buf Descriptors");
+        return;
+    }
+    base_DS.rx_buf_desc_start_addr = (volatile enet_rx_bd_struct_t *) ENET_ALIGN(memptr,
+                                                        ENET_BUFF_ALIGNMENT);
+    memset((enet_rx_bd_struct_t *) base_DS.rx_buf_desc_start_addr, 0,
+           sizeof(enet_rx_bd_struct_t) * ENET_RX_RING_LEN);
+
+    /* Allocating buffers for each RX Buffer Descriptor */
+    for (int i = 0; i < ENET_RX_RING_LEN; i++) {
+        base_DS.rx_buf[i] = MEM_ALLOC(
+        ENET_ETH_MAX_FLEN + ENET_BUFF_ALIGNMENT);
+        if (base_DS.rx_buf[i] == NULL) {
+            tr_error("Could not allocate memory for RX Buffers");
+            return;
+        }
+        base_DS.rx_buf[i] = (uint8_t*) ENET_ALIGN(base_DS.rx_buf[i],
+                                                  ENET_BUFF_ALIGNMENT);
     }
 
-    else if (link_status && last_known_status==LINK_STATUS_DOWN) {
-            last_known_status = LINK_STATUS_UP;
-            ethernet_iface_ptr->isInitialized = true;
-            tr_debug("Ethernet phy link status = up");
-            tr_info("Ethernet Cable plugged.");
-            enet_phy_link_setup(ethernet_iface_ptr);
+    /* Preparing the Buffer Configurations */
+    enet_buffer_config_t buf_config = { .rxBdNumber = ENET_RX_RING_LEN,
+        .txBdNumber = ENET_TX_RING_LEN, .rxBuffSizeAlign = ENET_ALIGN(
```

Comment:
What on earth happened to this structure? Did you press Shift-Control-F?

The formatter's not that intelligent - don't let it just trash your layout.